### PR TITLE
Tutorial and Editor component cypress test

### LINKED ISSUE #168: Write cypress tests for frontend components
Write cypress tests for frontend components

### PR DESCRIPTION
**Pull request by rijusougata13**
## Description

Tutorial and Editor component cypress test
## Related Issue

Fixed #168 
## Screenshots or GIF (In case of UI changes):
simplescreenrecorder-2021-08-04_15 39 31

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by rijusougata13** [comment]
What I have done :
1. Check for both editor and tutorial page is not accessable by non-loggedin user
2. Visit tutorial page and check if tutorial card, body, search , components are avaible
3. check when we click the button new tutorial modal open
4. check form has atleast required field present in it.
5. check all the buttons works as it is.
6. In the edior test we checked if all the components present 
7. check all the components are hidden before clicking the button and it should be visible after clicking button
8. check add image portal open or not

**Comment by thisuraseniya** [comment]
Pls fix merge conflicts

**Comment by rijusougata13** [comment]
> Pls fix merge conflicts

Fixed

**Review by Niweera — APPROVED** [review]

File: cypress/fixtures/login.json [hunk 1]
```diff
@@ -1,4 +1,4 @@
 {
-  "email": "rijusougata13@gmail.com",
-  "password": "12345678"
+  "email": "sougatariju13@gmail.com",
+  "password": "123456"
```
Comment (shehand) [inline cypress/fixtures/login.json:3]: if we change the firebase project, will this fail the tests? If so, this is not the correct approach then.

Comment (rijusougata13) [reply cypress/fixtures/login.json:3]: no tests are dependent on firebase components  other than dashboard component.firebase project change will impact only on dashboard components tests. Thats why I left only dashboard tests blank to discuss further( mentioned on #171). All the other tests is based on frontend like components rendering or not ,edit form working or not , buttons working as expected or not, non logged in user can access the site or not like this.

**Comment by thisuraseniya** [comment]
Another merge conflict :D

**Comment by rijusougata13** [comment]
fixing those conflicts.

**Comment by rijusougata13** [comment]
Fixed conflict :)

**Review by thisuraseniya — APPROVED** [review]

**Pull request merged**